### ISSUE TITLE
Can Oracle-front running be a problem to the pool ?

### ISSUE THREAD

**Issue by theDefiBat** [askibin/perpetuals#4]
need to recheck on the oracle front running issues , which GMX overcomes by having two step process to open and close positions.
Although the oracle latency is low on Solana, and with a good fee modal this can be handled , but still need to see if there are any other ways this gap can be exploited.

**Comment by askibin** [comment]
Besides fast oracles, there are some other built-in mechanisms in the program aimed at mitigating the risk:
- entry/exit prices are computed based on the worst of two: oracle price and EMA of the oracle price, plus there is a confidence interval check, which means that it won't be possible to game the protocol with relatively short sudden price swings.
- in addition to the fees, there is a configurable spread per token. This spread can be set to something like 2 stdev of price diffs between oracle updates. Spread might not even be necessary given the EMA-based pricing and if fees are high enough.
- last update time is checked for the oracle prices. It can be set to a very small number to avoid positions opening using stale prices

Of course, none of the above will help in the situation when a thin market is manipulated for a long period of time on centralized exchanges. I.e. when the price stays wacky longer than the EMA period. You can build an off-chain tool that would analyze the current state of the market from a high-level perspective and pause affected pools in periods of instability. E.g. if the price moves by X% in period Y, then trading is halted. The rules should be strictly defined and public, so there are no surprises to traders.

**Comment by mzs-dev** [comment]
Absolutely, the sweet spot lies between optimsing the execution for traders and mitigating risk for the pool...
Pausing the affected pools based on a given set of publicly known pre-conditions does seem to be better alterantive to having any other admin controlled action coming into play otherwise.
